hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseWriteHelper.java

```diff
@@ -87,6 +88,11 @@ public I deduplicateRecords(
     return deduplicateRecords(records, table.getIndex(), parallelism, table.getConfig().getSchema(), table.getConfig().getProps(), recordMerger);
   }
 
-  public abstract I deduplicateRecords(
-      I records, HoodieIndex<?, ?> index, int parallelism, String schema, Properties props, HoodieRecordMerger merge);
+  public I deduplicateRecords(
+      I records, HoodieIndex<?, ?> index, int parallelism, String schema, Properties props, HoodieRecordMerger merger) {
+    return innerDeduplicateRecords(records, index, parallelism, schema, HoodieAvroRecordMerger.withDeDuping(props), merger);
```

Comment:
It's better to actually do the other way around:

 - Keep inherited method named `deduplicateRecords`
 - Create new _private_ method `deduplicateRecordsInternal` modifying props and invoke it in this class

Reply:
Well, deduplicateRecords also used in StreamWriteFunction in Flink. So we should modify props in deduplicateRecords.